contracts/modules/cover/Cover.sol

```diff
@@ -781,20 +807,26 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard {
     return uint(activeCover[assetId].totalActiveCoverInAsset);
   }
 
-  function isPoolAllowed(uint productId, uint poolId) external view returns (bool) {
+  // Returns true if the product exists and the pool is authorized to have the product
+  function isPoolAllowed(uint productId, uint poolId) public view returns (bool) {
 
-    uint poolCount = allowedPools[productId].length;
-
-    if (poolCount == 0) {
-      return true;
-    }
+    // If product exists and is not deprecated, check if the pool is allowed
+    if (doesProductExist(productId) && !isProductDeprecated(productId)) {
 
-    for (uint i = 0; i < poolCount; i++) {
-      if (allowedPools[productId][i] == poolId) {
+      uint poolCount = allowedPools[productId].length;
+      // If no pools are specified, nothing is blacklisted
+      if (poolCount == 0) {
         return true;
       }
-    }
 
+      for (uint i = 0; i < poolCount; i++) {
+        if (allowedPools[productId][i] == poolId) {
+          // If allowed pools are specified, the product must already be initialized
+          return true;
+        }
+      }
+    }
+    // product doesn't exist or is deprecated
```

Comment:
let's fix the comment so it matches what happens

basically the pool is not supported